src/api/routes/address.ts

```diff
@@ -76,32 +72,58 @@ async function getBlockHeight(
   return blockHeight;
 }
 
-interface AddressAssetEvents {
-  results: TransactionEvent[];
-  limit: number;
-  offset: number;
-  total: number;
-}
-
-export function createAddressRouter(db: PgStore, chainId: ChainID): express.Router {
-  const router = express.Router();
-  const cacheHandler = getETagCacheHandler(db);
-  const mempoolCacheHandler = getETagCacheHandler(db, ETagType.mempool);
-
-  router.get(
-    '/:stx_address/stx',
-    cacheHandler,
-    asyncHandler(async (req, res, next) => {
-      const stxAddress = req.params['stx_address'];
+export const AddressRoutes: FastifyPluginAsync<
+  Record<never, never>,
+  Server,
+  TypeBoxTypeProvider
+> = async fastify => {
+  fastify.get(
+    '/:principal/stx',
+    {
+      preHandler: handleChainTipCache,
+      schema: {
+        operationId: 'get_account_stx_balance',
+        summary: 'Get account STX balance',
+        description: `Retrieves STX token balance for a given Address or Contract Identifier.`,
+        tags: ['Accounts'],
+        params: Type.Object({
+          principal: Type.String({
+            description: 'Stacks address or a Contract identifier',
+            examples: ['SP31DA6FTSJX2WGTZ69SFY11BH51NZMB0ZW97B5P0'],
+          }),
```

Comment:
I think we can reuse a simple `AddressSchema` or similar in a bunch of these endpoints

Reply:
Good catch